lib/tasks/control_search_requests.rake

```diff
@@ -0,0 +1,19 @@
+namespace :db do
+  MAX_REQUESTS = 1000
+  STORAGE_DAYS = 10
+
+  desc 'Limits the requests in the database to the most recent and limit the total number'
+  task manage_requests: :environment do
+    total_destroyed = 0
+    # Delete requests over STORAGE_DAYS days old
+    total_destroyed += Request.where('created_at < ?', STORAGE_DAYS.days.ago).destroy_all.length
+
+    # Ensure we have no more than MAX_REQUESTS of the most recent requests
+    if Request.count > MAX_REQUESTS
+      total_destroyed += Request.order(id: :desc).offset(Request.count - MAX_REQUESTS).destroy_all.length
```

Comment:
A couple of changes to suggest:

1. You maybe want to order by `created_at`
2. Can't you just `offset(MAX_REQUESTS)`